icepyx/core/APIformatting.py

```diff
@@ -181,12 +184,56 @@ def to_string(params):
     return "&".join(param_list)
 
 
+ParameterType = Literal["CMR", "required", "subset"]
+# DevGoal: When Python 3.12 is minimum supported version, migrate to PEP695 style
+T = TypeVar("T", contravariant=False, bound=ParameterType)
+
+
+class _FmtedKeysDescriptor:
+    """Enable the Parameters class' fmted_keys property to be typechecked correctly.
+
+    See: https://github.com/microsoft/pyright/issues/3071#issuecomment-1043978070
+    """
+
+    @overload
+    def __get__(  # type: ignore
+        self,
+        instance: 'Parameters[Literal["CMR"]]',
```

Comment:
This allows the return type of `fmted_keys` method to depend on the `partype` the class is initialized with

Reply:
Woops, commented one line too low

Reply:
Actually... it shows up as one line too low in the diff view, but it shows up in the right place in the conversation view. Anyone else seeing that?